README.md

```diff
@@ -83,6 +105,14 @@ if (map.has(key)) {
 }
 ```
 
+With this proposal:
+
+```js
+map.upsert(key, old => old.doThing()).doThing();
```

Comment:
I'm confused why call `doThing()` again.

Reply:
miswrote the fix, should not have the extra `.doThing()`

Reply:
```suggestion
map.upsert(key, old => old.doThing())
```